prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -2164,16 +2169,49 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
                             debug!("Encoding call of PartialEq::ne");
                             stmts.extend(
                                 self.encode_cmp_function_call(
-                                    def_id,
-                                    location,
-                                    term.source_info.span,
-                                    args,
-                                    destination,
-                                    vir::BinOpKind::NeCmp,
+                                def_id,
+                                location,
+                                term.source_info.span,
+                                args,
+                                destination,
+                                vir::BinOpKind::NeCmp,
                                 )?
                             );
                         }
 
+                        "core::f32::<impl f32>::is_nan" |
+                        "core::f64::<impl f64>::is_nan" => {
+                            let span = self.mir_encoder.get_span_of_location(location);
+                            let operand = self.mir_encoder.encode_operand_expr(&args[0]).with_span(span)?;
+                            let operand_box = std::boxed::Box::from(operand);
+                            let expr = vir::Expr::UnaryOp(UnaryOpKind::IsNaN, operand_box, vir::Position::default());
```

Comment:
Assuming `Expr::is_nan` was added:
```suggestion
                            let expr = vir::Expr::is_nan(operand);
```